lib/archethic_web/graphql_schema/resolver.ex

```diff
@@ -18,6 +18,30 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
 
   @limit_page 10
 
+  def get_genesis_address(address) do
+    t1 = Task.async(fn -> TransactionChain.fetch_genesis_address_remotely(address) end)
```

Comment:
Hello, ArchethicWeb should be agnostic to the underlying architecture of the archethic project ( it should only interact with the public api in the Archethic module).
I would suggest to change this call to `Archethic.fetch_genesis_address_remotely`

Reply:
you probably don't need a Task here, I would call directly the function in this case.

Reply:
> Hello, ArchethicWeb should be agnostic to the underlying architecture of the archethic project ( it should only interact with the public API in the Archethic module).

Ok thanks, I'll fix that! Could you explain to me why it should be agnostic to the underlying architecture of the archethic project, I don't really understand?

Reply:
ArchethicWeb represents the frontend and Archethic represents the backend logic, in here they are in the same repository but this can change someday in the future, also we try to follow the hexagonal architecture (some call it clean architecture).

Finally if the web only depends on the public api, it means that if we refactor the whole backend tomorrow we will not need to change the web part and separate people / teams can work independently on different parts of the project. 

Feel free to ping me if it isn't clear.

Reply:
Very clear, thanks for the explanation!

I tried using the call `Archethic.fetch_genesis_address_remotely` as you suggested however it seems to always return the same address as entered.

here is the code:
```
def get_genesis_address(address) do
    case Archethic.fetch_genesis_address_remotely(address) do
      {:ok, genesis_address} ->
        {:ok, %{genesis: genesis_address}}

      _ ->
        {:ok, %{genesis: address}}
    end
end
```
Correcting the code by using TransactionChain.fetch_genesis_address_remotely(address) fixes the issue.
Do you have any idea why or what I might be doing wrong?

Reply:
you can look at the definition here: https://github.com/archethic-foundation/archethic-node/blob/develop/lib/archethic.ex#L246